Makefile.in

```diff
@@ -371,29 +383,48 @@ SINGA_HDRS := include/singa.h \
 
 lib_LTLIBRARIES = libsinga.la
 libsinga_la_SOURCES = $(PROTO_HDRS) $(PROTO_SRCS) $(SINGA_HDRS) $(SINGA_SRCS)
-libsinga_la_CXXFLAGS = -O3 -Wall -pthread -fPIC -std=c++11 -Wno-unknown-pragmas \
-                       $(MSHADOW_FLAGS) -DCPU_ONLY=1 -funroll-loops -gdwarf-2 \
-                       -gstrict-dwarf -Woverloaded-virtual -DTHREADED -fpermissive
+@LMDB_FALSE@libsinga_la_CXXFLAGS = -O3 -Wall -pthread -fPIC -std=c++11 -Wno-unknown-pragmas \
+@LMDB_FALSE@                       $(MSHADOW_FLAGS) -DCPU_ONLY=1 -funroll-loops -gdwarf-2 \
+@LMDB_FALSE@                       -gstrict-dwarf -Woverloaded-virtual -DTHREADED -fpermissive
+
+@LMDB_TRUE@libsinga_la_CXXFLAGS = -O3 -Wall -pthread -fPIC -std=c++11 -Wno-unknown-pragmas \
+@LMDB_TRUE@                       $(MSHADOW_FLAGS) -DCPU_ONLY=1 -funroll-loops -gdwarf-2 \
+@LMDB_TRUE@                       -gstrict-dwarf -Woverloaded-virtual -DTHREADED -fpermissive -DUSE_LMDB
 
 libsinga_la_LDFLAGS = -I./include
 singa_SOURCES = src/main.cc
 singa_CXXFLAGS = -O3 -Wall -pthread -fPIC -std=c++11 -MMD -Wno-unknown-pragmas \
                  $(MSHADOW_FLAGS) -DCPU_ONLY=1 -funroll-loops -DTHREADED
 
-singa_LDFLAGS = -I./include \
-                -lsinga \
-                -lglog  \
-                -lgflags \
-                -lprotobuf \
-                -lrt \
-                -lopencv_highgui \
-                -lopencv_imgproc \
-                -lopencv_core \
-                -lopenblas \
-                -lzmq \
-                -lczmq \
-                -llmdb \
-                -lzookeeper_mt
+@LMDB_FALSE@singa_LDFLAGS = -I./include \
```

Comment:
can simplify this by including all necessary libs, and then including optional libs if configured?